### PR TITLE
LPS-185464 Allow FDS Filters to be set with `null` value

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-185464 Fix all the things](https://issues.liferay.com/browse/LPS-185464)

### What is the goal of this PR?

This issue came up during development of FDS fragments, and it is not the first time I'm seeing it. The problem is that if we are setting a default through `defaultProps`, or a React default on a prop, it will end up overridden with a `null` value set in backend or parent component. As it works fine with `undefined` or an empty array, what we previously did is omitted setting the prop, or made sure backend sets an empty array as default. This ended up being proven bad approach, as it keeps resulting in bugs. So, in this PR, we are allowing setting of `null` value, with the behavior equal to `undefined` and `[]`.

### What does it look like?

No changes in UI.

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js [hunk 1]
```diff
@@ -151,7 +151,9 @@ const FrontendDataSet = ({
 			...initialActiveView,
 		};
 
-		const filters = initialFilters.map((filter) => {
+		const filters = [];
+
+		initialFilters?.forEach((filter) => {
```
Comment (bryceosterhaus) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js:156]: How come you switched from map to a forEach+push? (I typically find map easier to reason through than a forEach+push.

For checking null, you could do `initialFilters ? initialFilters.map(... : []`

Comment (bryceosterhaus) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js:156]: ^ non blocker fyi

**Review by bryceosterhaus — APPROVED** [review]

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js:156]: Yeah, that looks better, I'll make the change. `forEach` somewhat implies splitting of the array, where here we modify and return the same number of items.

**Comment by markocikos** [comment]
ci:forward

**Review by kresimir-coko — APPROVED** [review]

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**